src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java

```diff
@@ -29,32 +33,60 @@
 import org.eclipse.lsp4j.ServerCapabilities;
 import org.eclipse.lsp4j.TextDocumentIdentifier;
 import org.eclipse.lsp4j.TextDocumentSyncKind;
+import org.eclipse.lsp4j.WorkspaceFolder;
 import org.eclipse.lsp4j.services.LanguageClient;
 import org.eclipse.lsp4j.services.LanguageClientAware;
 import org.eclipse.lsp4j.services.LanguageServer;
 import org.eclipse.lsp4j.services.TextDocumentService;
 import org.eclipse.lsp4j.services.WorkspaceService;
 
+import software.amazon.smithy.lsp.ext.LspLog;
+import software.amazon.smithy.lsp.ext.SmithyBuildExtensions;
+import software.amazon.smithy.lsp.ext.ValidationException;
+
 public class SmithyLanguageServer implements LanguageServer, LanguageClientAware, SmithyProtocolExtensions {
 
   private Optional<LanguageClient> client = Optional.empty();
+  private Optional<SmithyBuildExtensions> ext = Optional.empty();
 
   private File workspaceRoot = null;
 
-  private SmithyTextDocumentService tds = null;
+  private Optional<SmithyTextDocumentService> tds = Optional.empty();
 
   @Override
   public CompletableFuture<Object> shutdown() {
     return Utils.completableFuture(new Object());
   }
 
+  private void loadSmithyBuild(File root) throws ValidationException, FileNotFoundException {
+    File smithyBuild = Paths.get(root.getAbsolutePath(), "smithy-build.json").toFile();
```

Comment:
AWS's intellij plugin assumes [these config files](https://github.com/awslabs/smithy-intellij/blob/a4a935a619dbb5662599bedaa713c64304854b6b/src/main/java/software/amazon/smithy/plugin/SmithyPostStartupActivity.java#L46-L49) as well (and our current vscode extension assumes smithy.json). We could simply look for all of those and merge them together (make `SmithyBuildExtensions` a "monoid"-like construct for instance)